pkg/reconciler/cronjobsource/env_lookup.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronjobsource
+
+import (
+	"fmt"
+	"os"
+	"sync"
+)
+
+// EnvLookup returns the value of the associated environment variable. It only does the lookup once.
+// If the environment variable is undefined, then every call to GetValue() will panic.
+type EnvLookup interface {
```

Comment:
why not using `"github.com/kelseyhightower/envconfig"?

Reply:
Good point, switched. Note that this does change the failure point from when reconciling a CronJobSource to process startup.